dev-tools/build_release.py

```diff
@@ -618,6 +610,60 @@ def check_norelease(path='src'):
           if pattern.search(line):
             raise RuntimeError('Found //norelease comment in %s line %s' % (full_path, line_number))
 
+def run_and_print(text, run_function):
+  try:
+    print(text, end='')
+    run_function()
+    print(OKGREEN + 'OK' + ENDC)
+    return True
+  except RuntimeError:
+    print(FAIL + 'NOT OK' + ENDC)
+    return False
+
+def check_env_var(text, env_var):
+  try:
+    print(text, end='')
+    env[env_var]
+    print(OKGREEN + 'OK' + ENDC)
+    return True
+  except KeyError:
+    print(FAIL + 'NOT OK' + ENDC)
+    return False
+
+def check_environment_and_commandline_tools(check_only):
+  checks = list()
```

Comment:
Hmm can't do `ok and ...` because Python will short-circuit and won't print the failures...

Maybe just:

```
if check_env_var(...):
  ok = False
```

for each check?

Or leave it as a list :)